lib/browser/api/menu.js

```diff
@@ -5,262 +5,115 @@ const EventEmitter = require('events').EventEmitter
 }
 
+// close an existing popup
 Menu.prototype.closePopup = function (window) {
-  if (window == null || window.constructor !== BrowserWindow) {
+  if (!window || window.constructor !== BrowserWindow) {
     window = BrowserWindow.getFocusedWindow()
   }
-  if (window != null) {
-    this.closePopupAt(window.id)
-  }
+  if (window) this.closePopupAt(window.id)
 }
 
+// return a MenuItem with the specified id
 Menu.prototype.getMenuItemById = function (id) {
   const items = this.items
 
   let found = items.find(item => item.id === id) || null
-  for (let i = 0, length = items.length; !found && i < length; i++) {
+  for (let i = 0; !found && i < items.length; i++) {
     if (items[i].submenu) {
       found = items[i].submenu.getMenuItemById(id)
     }
   }
   return found
 }
 
+// append a new MenuItem to an existing Menu
 Menu.prototype.append = function (item) {
   return this.insert(this.getItemCount(), item)
 }
 
+// insert a new menu item at a specified location
 Menu.prototype.insert = function (pos, item) {
-  var base, name
-  if ((item != null ? item.constructor : void 0) !== MenuItem) {
+  if ((item ? item.constructor : void 0) !== MenuItem) {
     throw new TypeError('Invalid item')
   }
-  switch (item.type) {
-    case 'normal':
-      this.insertItem(pos, item.commandId, item.label)
-      break
-    case 'checkbox':
-      this.insertCheckItem(pos, item.commandId, item.label)
-      break
-    case 'separator':
-      this.insertSeparator(pos)
-      break
-    case 'submenu':
-      this.insertSubMenu(pos, item.commandId, item.label, item.submenu)
-      break
-    case 'radio':
-      // Grouping radio menu items.
-      item.overrideReadOnlyProperty('groupId', generateGroupId(this.items, pos))
-      if ((base = this.groupsMap)[name = item.groupId] == null) {
-        base[name] = []
-      }
-      this.groupsMap[item.groupId].push(item)
 
-      // Setting a radio menu item should flip other items in the group.
-      v8Util.setHiddenValue(item, 'checked', item.checked)
-      Object.defineProperty(item, 'checked', {
-        enumerable: true,
-        get: function () {
-          return v8Util.getHiddenValue(item, 'checked')
-        },
-        set: () => {
-          var j, len, otherItem, ref1
-          ref1 = this.groupsMap[item.groupId]
-          for (j = 0, len = ref1.length; j < len; j++) {
-            otherItem = ref1[j]
-            if (otherItem !== item) {
-              v8Util.setHiddenValue(otherItem, 'checked', false)
-            }
-          }
-          return v8Util.setHiddenValue(item, 'checked', true)
-        }
-      })
-      this.insertRadioItem(pos, item.commandId, item.label, item.groupId)
-  }
-  if (item.sublabel != null) {
-    this.setSublabel(pos, item.sublabel)
-  }
-  if (item.icon != null) {
-    this.setIcon(pos, item.icon)
-  }
-  if (item.role != null) {
-    this.setRole(pos, item.role)
-  }
+  // insert item depending on its type
```

Comment:
I'm not sure what our comment policy is, but I personally don't think comments like these are helpful. This particular comment is just rephrasing the line below in English rather than JS.

I think comments should describe *why* this the code is there (if needed) rather than *what* the code does. The latter should be obvious from reading the code although there are of course exceptions.

What do you think?

Reply:
yeah, i think you're right :) it's too common sense to be neccessary

Reply:
Thanks for fixing 👍